.circleci/config.yml

```diff
@@ -66,11 +54,13 @@ jobs:
           at: ~/app
       - setup_remote_docker:
           docker_layer_caching: true
+      - mem/recall:
+          env_var: APP_VERSION
       - run:
           name: Build container and publish to docker hub
           command: |
-            APP_VERSION=$(cat ./app_version.txt)
             docker build -f docker/Dockerfile.aws --rm=false . \
+               --build-arg BUILD_NUMBER=$APP_VERSION \
```

Comment:
if you do it this way then you'll need to change the Dockerfile to be a multi stage docker file instead - see auth project for example.  Also then don't need to do a `gradle build`, can do a `gradle test` instead

Reply:
Cool, thanks.  I wonder if it would be good to make the Gradle plugin generate the Dockerfile (and also run.sh) - would take more duplication out of the build.